### PR TITLE
Feature/lanelet2 install

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description
Draft PR showing a buildable installation for lanelet2 and lanelet2_extension

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by paulbourelly999 — COMMENTED** [review]
Just had a couple questions about these changes

File: examples/Dockerfile [hunk 1]
```diff
@@ -59,6 +62,69 @@ RUN cmake ..
...
+# Install PROJ, a package for coordinate transformations
+RUN sudo git clone https://github.com/OSGeo/PROJ.git /home/carma-streets-ws/PROJ --branch 6.2.1 && \
+        cd /home/carma-streets-ws/PROJ && \
+        sudo ./autogen.sh && \
+        sudo ./configure && \
+        sudo make && \
+        sudo make install
+        
+# Download a cmake module for PROJ
+RUN cd /usr/share/cmake-3.10/Modules && sudo curl -O https://raw.githubusercontent.com/mloskot/cmake-modules/master/modules/FindPROJ4.cmake
```
[the remark this answers is not in the document]
Comment (msmcconnell) [reply examples/Dockerfile:74]: Adding it to this directory is effectively putting it in the root path for CMAKE_MODULE_PATH. So you shouldn't need to do anything extra for it.

File: examples/Dockerfile [hunk 2]
```diff
@@ -59,6 +62,69 @@ RUN cmake ..
...
+
+SHELL ["/bin/bash", "-c"]
+
+# Install CARMA Ready Lanelet2
+# Lanelet2 is written on the assumption ROS exists. Therefore some special steps are needed for compilation
+# First catkin_make must be installed. The version needed is only available with the ros-melodic-catkin build
+# This means the ros source list must be added to sudo to allow this installation
+# Additionally, the python-rospkg package managment tool is required for catkin to work properly 
+RUN sudo DEBIAN_FRONTEND=noninteractive apt update && \
+    sudo DEBIAN_FRONTEND=noninteractive apt install -y ros-melodic-catkin && \
```
[the remark this answers is not in the document]
Comment (msmcconnell) [reply examples/Dockerfile:92]: DEBIAN_FRONTEND=noninteractive 
Is needed for certain kinds of prompts that don't conform to the (yes/no) type structure. In this case, catkin wanted me to select my timezone by entering a number which didn't play well with -y.

File: examples/Dockerfile [hunk 3]
```diff
@@ -59,6 +62,69 @@ RUN cmake ..
...
+WORKDIR  /home/carma-streets-ws/carma_lanelet2
+
+# In order to trick lanelet2 into building the ROS_VERSION environment variable must be set
+# In order to fully decouple lanelet2_extension from ros the LANELET2_EXTENSION_LOGGER_TYPE environment variable must be set
+RUN source /opt/ros/melodic/setup.bash && \
+    sudo DEBIAN_FRONTEND=noninteractive apt-get install -y python-rospkg && \
+    sudo DEBIAN_FRONTEND=noninteractive apt-get install -y libeigen3-dev && \
+    ROS_VERSION=1 LANELET2_EXTENSION_LOGGER_TYPE=1 catkin_make install
+
+
```
[the remark this answers is not in the document]
Comment (msmcconnell) [reply examples/Dockerfile:126]: 2 steps.
Frist, ensure the installation is on your PATH, easiest way is 
```
source /home/carma-streets-ws/carma_lanelet2/install/setup.bash
```
You might be able to copy the binaries out as well if needed. 

Second, you would need to use find_package in your CMakeLists.txt file. I sent you an email a month or two ago with instructions for that using catkin. It should be the same process.

**Comment by paulbourelly999** [comment]
This was a example PR to show including lanelet2 dependencies in docker image. Not necessary for merge

**Pull request closed without merging**